mesonbuild/minstall.py

```diff
@@ -441,6 +446,25 @@ def do_copyfile(self, from_file: str, to_file: str,
         append_to_log(self.lf, to_file)
         return True
 
+    def do_symlink(self, target: str, link: str, full_dst_dir: str) -> bool:
+        abs_target = target
+        if not os.path.isabs(target):
+            abs_target = os.path.join(full_dst_dir, target)
+        if not os.path.exists(abs_target):
+            raise RuntimeError(f'Tried to install symlink to missing file {abs_target}')
+        if os.path.exists(link):
+            link_target = os.readlink(link)
+            if link_target == target:
+                self.log(f'Symlink {link} to {target} already exists, skipping')
+                return True
+            else:
+                # What should we do in this case? Should we consider it?
+                return False
+        self.log(f'Installing symlink {link} to {target}')
+        os.symlink(target, link, target_is_directory=os.path.isdir(abs_target))
```

Comment:
pathlib.Path has a function for creating symlinks.

Reply:
And we shouldn't use that!

Instead, we should use self.symlink, which handles --dry-run, and may choose to do pathlib stuff if it wants to.

Reply:
Thanks! Moved to self.symlink.